### PR TITLE
Added LoginTask and Updated Execution graph when While,If,Switch logic.

### PR DESCRIPTION
**Pull request by atimhome**
Demo video
While
https://www.youtube.com/watch?v=SCulwAp6u2U
If
https://www.youtube.com/watch?v=wYye3IIUCHU
Switch
https://www.youtube.com/watch?v=kSZpAQiEwZI

### CODE REVIEW

**Comment by alexhiggins732** [comment]
Where are there Chinese characters embedded in comments in the Assembly files?

File: src/dotnet/Wexflow.Tasks.WhileCheck/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
+﻿using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+// 組件的一般資訊是由下列的屬性集控制。
```
[the remark this answers is not in the document]
Comment (atimhome) [reply src/dotnet/Wexflow.Tasks.WhileCheck/Properties/AssemblyInfo.cs]: Yes, you can.

File: src/dotnet/Wexflow.Tasks.SwitchCheck/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
+﻿using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+// 組件的一般資訊是由下列的屬性集控制。
```
[the remark this answers is not in the document]
Comment (atimhome) [reply src/dotnet/Wexflow.Tasks.SwitchCheck/Properties/AssemblyInfo.cs]: Yes, you can.

File: src/dotnet/Wexflow.Tasks.MessageCorrect/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
+﻿using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+// 組件的一般資訊是由下列的屬性集控制。
```
[the remark this answers is not in the document]
Comment (atimhome) [reply src/dotnet/Wexflow.Tasks.MessageCorrect/Properties/AssemblyInfo.cs]: Yes, you can.

File: src/dotnet/Wexflow.Tasks.IfCheck/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
+﻿using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+// 組件的一般資訊是由下列的屬性集控制。
```
[the remark this answers is not in the document]
Comment (atimhome) [reply src/dotnet/Wexflow.Tasks.IfCheck/Properties/AssemblyInfo.cs]: Yes, you can.

File: src/dotnet/Wexflow.Core/WexflowGraph.cs [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+                }
+
+                // Get the last nodes id,to Combine the switch nodes to next node
+                nextNodeParentId = this._GetLastNodeID(switchConditionId, switchCase.Nodes, switchCase.Value, nextNodeParentId);
+            }
+           
+            // Remove first "&"
+            nextNodeParentId = (nextNodeParentId.Length > 0) ? nextNodeParentId.Remove(0, 1) : nextNodeParentId;
+
+            // Add Switch Condition Node(加入Switch判斷式的節點)
```
[the remark this answers is not in the document]
Comment (atimhome) [reply src/dotnet/Wexflow.Core/WexflowGraph.cs]: // Add Switch Condition Node(加入Switch判斷式的節點)
=>
// Add Switch Condition Node

File: src/dotnet/Wexflow.Core/WexflowGraph.cs [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+        }
+
+        private IList<GraphNode> _GetExecutionGraphWhileNode(Node node, out string nextNodeParentId, string parentLayer = "", string selfLayer = "")
+        {
+            IList<GraphNode> dataList = new List<GraphNode>();
+            string layer = parentLayer + "n";
+            nextNodeParentId = string.Empty;
+            While whileNode = node as While;
+
+            // Add While Condition Node(加入While判斷式的節點)
```
[the remark this answers is not in the document]
Comment (atimhome) [reply src/dotnet/Wexflow.Core/WexflowGraph.cs]: // Add While Condition Node(加入While判斷式的節點)
=>
// Add While Condition Node

File: src/dotnet/Wexflow.Core/WexflowGraph.cs [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+            }
+
+            // Get the last nodes id,to Combine the if nodes to next node            
+            nextNodeParentId = this._GetLastNodeID(ifConditionId, ifNode.DoNodes, "Do", nextNodeParentId);
+            nextNodeParentId = this._GetLastNodeID(ifConditionId, ifNode.ElseNodes, "Else", nextNodeParentId);
+
+            // Remove first "&"
+            nextNodeParentId = (nextNodeParentId.Length > 0) ? nextNodeParentId.Remove(0, 1) : nextNodeParentId;
+
+            // Add IF Condition Node(加入IF判斷式的節點)
```
[the remark this answers is not in the document]
Comment (atimhome) [reply src/dotnet/Wexflow.Core/WexflowGraph.cs]: // Add IF Condition Node(加入IF判斷式的節點)
=>
// Add IF Condition Node

File: src/dotnet/Wexflow.Core/WexflowGraph.cs [hunk 8]
```diff
@@ -0,0 +1,271 @@
...
+        private IList<GraphNode> _GetExecutionGraphIfNode(Node node, out string nextNodeParentId, string parentLayer = "", string selfLayer = "")
+        {
+            IList<GraphNode> dataList = new List<GraphNode>();
+            string layer = parentLayer + "n";
+            nextNodeParentId = string.Empty;
+            If ifNode = node as If;
+
+            string ifConditionId = layer + ifNode.Id + selfLayer;
+
+            // Add DO and Else Nodes(加入DO跟Else的節點)
```
[the remark this answers is not in the document]
Comment (atimhome) [reply src/dotnet/Wexflow.Core/WexflowGraph.cs]: // Add DO and Else Nodes(加入DO跟Else的節點)      
=>
// Add DO and Else Nodes

File: src/dotnet/Wexflow.Core/WexflowGraph.cs [hunk 9]
```diff
@@ -0,0 +1,271 @@
...
+                    dataList = dataList.Concat(this._GetExecutionGraphNode(node, out lastConditionNodeId, parentLayer, selfLayer)).ToList();
+                    continue;
+                }
+
+                List<string> parentIdList = lastConditionNodeId.Split('&').ToList();
+                lastConditionNodeId = string.Empty;
+                node.ParentId = -2;
+                dataList = dataList.Concat(this._GetExecutionGraphNode(node, out lastConditionNodeId, parentLayer, selfLayer)).ToList();
+
+                // 將if else的線拉回至此節點
```
[the remark this answers is not in the document]
Comment (atimhome) [reply src/dotnet/Wexflow.Core/WexflowGraph.cs]: // 將if else的線拉回至此節點
=>
// pull the 「if else」's line  to this node.

File: src/dotnet/Wexflow.Core/WexflowGraph.cs [hunk 10]
```diff
@@ -0,0 +1,271 @@
...
+                    dataList.Add(new GraphNode(dataList.Last().Id, dataList.Last().Name, pid));
+                }
+            }
+
+            // 最後一個節點判斷
+            if (string.IsNullOrWhiteSpace(lastConditionNodeId) == false)
+            {
+                // 將資料送到上一層
+                childParentId = lastConditionNodeId;
+            }
```
[the remark this answers is not in the document]
Comment (atimhome) [reply src/dotnet/Wexflow.Core/WexflowGraph.cs]: // 將資料送到上一層
=>
// Send data to upper layer.

File: src/dotnet/Wexflow.Core/WexflowGraph.cs [hunk 11]
```diff
@@ -0,0 +1,271 @@
...
+        }
+
+        private IList<GraphNode> _GetExecutionGraphSwitchNode(Node node, out string nextNodeParentId, string parentLayer = "", string selfLayer = "")
+        {
+            IList<GraphNode> dataList = new List<GraphNode>();
+            string layer = parentLayer + "n";
+            nextNodeParentId = string.Empty;
+            Switch switchNode = node as Switch;
+
+            // Add Cases Nodes(加入Cases的節點)
```
[the remark this answers is not in the document]
Comment (atimhome) [reply src/dotnet/Wexflow.Core/WexflowGraph.cs]: // Add Cases Nodes(加入Cases的節點)
=>
// Add Cases Nodes

File: src/dotnet/Wexflow.Core/WexflowGraph.cs [hunk 12]
```diff
@@ -0,0 +1,271 @@
...
+            While whileNode = node as While;
+
+            // Add While Condition Node(加入While判斷式的節點)       
+            string whileConditionId = layer + whileNode.Id + selfLayer;            
+            string whileConditionParentId = (node.ParentId == -2) ? "n-1" : (node.ParentId == -1) ? (string.IsNullOrEmpty(parentLayer)) ? "n-1" : parentLayer : layer + whileNode.ParentId + selfLayer;
+            string whileConditionTaskName = this._GetTaskName(whileNode.WhileId);
+            string description = (whileConditionId.IndexOf(whileConditionParentId) == -1) ? string.Empty : selfLayer;
+            dataList.Add(new GraphNode(whileConditionId, whileConditionTaskName, whileConditionParentId, description));
+
+            // Add While Child Nodes(加入子節點)
```
[the remark this answers is not in the document]
Comment (atimhome) [reply src/dotnet/Wexflow.Core/WexflowGraph.cs]: // Add While Child Nodes(加入子節點)
=>
// Add While Child Nodes

**Pull request closed without merging**